src/gui/tray/ActivityItemContextMenu.qml

```diff
@@ -0,0 +1,45 @@
+import QtQml 2.15
+import QtQuick 2.15
+import QtQuick.Controls 2.3
+
+AutoSizingMenu {
+    id: moreActionsButtonContextMenu
+
+    property int maxActionButtons: 0
+
+    property var activityItemLinks: []
+
+    signal menuEntryTriggered(int index)
+
+    // transform model to contain indexed actions with primary action filtered out
+    function actionListToContextMenuList(actionList) {
+        // early out with non-altered data
+        if (actionList.length <= maxActionButtons) {
+            return actionList;
+        }
+
+        // add index to every action and filter 'primary' action out
+        var reducedActionList = actionList.reduce(function(reduced, action, index) {
+            if (!action.primary) {
+                var actionWithIndex = { actionIndex: index, label: action.label };
+                reduced.push(actionWithIndex);
+            }
+            return reduced;
+        }, []);
+
+
+        return reducedActionList;
+    }
```

Comment:
Seems to me that the job of the model is precisely to provide the data in the manner needed by the UI's view components, rather than having the view components perform a conversion of the model's data -- happy to hear what others think though :)